src/main/scala/firrtl/Compiler.scala

```diff
@@ -478,3 +479,6 @@ trait Compiler extends LazyLogging {
 
 }
 
+trait ProvidesOptions {
+  def provideOptions: (OptionParser[AnnotationSeq]) => Unit
```

Comment:
Is there anyway this could be immutable? Is it mutable because that's what the ExecutionOptions are using anyway?

Reply:
Also is there any reason it's `def provideOptions: (OptionParser[AnnotationSeq]) => Unit` instead of just `def provideOptions(parser: OptionParser[AnnotationSeq]: Unit`

Reply:
I may need some clarification on your first question about how this is mutable. Are you saying that this is able to mutably keep adding options to the parser if we're not disciplined and only call it once?

I'll have to check on the second point. It seems like I would have tried that first and there may have been a problem. However, it's possibly just over-engineered. I can test it.